libs/features/mas/web-components/src/variants/mini-compare-chart.js

```diff
@@ -0,0 +1,223 @@
+import { CSS } from './mini-compare-chart.css.js';
+import { DESKTOP_UP, TABLET_DOWN } from '../media.js';
+const FOOTER_ROW_MIN_HEIGHT = 32; // as per the XD.
+
+export class MiniCompareChart extends VariantLayout {
+  constructor(card) {
+    super(card);
+  }
+
+  #container;
+
+  getRowMinHeightPropertyName = (index) =>
+    `--consonant-merch-card-footer-row-${index}-min-height`;
+
+  getContainer() {
+    this.#container = this.#container ?? this.card.closest('[class*="-merch-cards"]') ?? this.card.parentElement;
+    return this.#container;
+  }
+
+  getGlobalCSS() {
+    return CSS;
+  }
+
+  getMiniCompareFooter = () => {
+    const secureLabel = this.card.secureLabel
+        ? html`<slot name="secure-transaction-label">
+              <span class="secure-transaction-label"
+                  >${this.card.secureLabel}</span
+              ></slot
+          >`
+        : html`<slot name="secure-transaction-label"></slot>`;
+    return html`<footer>${secureLabel}<slot name="footer"></slot></footer>`;
+  }
+
+  updateMiniCompareElementMinHeight (el, name) {
+    const elMinHeightPropertyName = `--consonant-merch-card-mini-compare-${name}-height`;
+    const height = Math.max(
+        0,
+        parseInt(window.getComputedStyle(el).height) || 0,
+    );
+    const maxMinHeight =
+        parseInt(
+            this.getContainer().style.getPropertyValue(elMinHeightPropertyName),
+        ) || 0;
+    if (height > maxMinHeight) {
+        this.getContainer().style.setProperty(
+            elMinHeightPropertyName,
+            `${height}px`,
+        );
+    }
+  }
+
+  adjustMiniCompareBodySlots () {
+    if (this.card.getBoundingClientRect().width === 0) return;
+  
+    this.updateMiniCompareElementMinHeight(
+        this.card.shadowRoot.querySelector('.top-section'),
+        'top-section',
+    );
+  
+    const slots = [
+        'heading-m',
+        'body-m',
+        'heading-m-price',
+        'price-commitment',
+        'offers',
+        'promo-text',
+        'callout-content',
+        'secure-transaction-label',
+    ];
+  
+    slots.forEach((slot) =>
+        this.updateMiniCompareElementMinHeight(
+            this.card.shadowRoot.querySelector(`slot[name="${slot}"]`),
+            slot,
+        ),
+    );
+    this.updateMiniCompareElementMinHeight(
+        this.card.shadowRoot.querySelector('footer'),
+        'footer',
+    );
+  
+    const badge = this.card.shadowRoot.querySelector(
+        '.mini-compare-chart-badge',
+    );
+    if (badge && badge.textContent !== '') {
+        this.getContainer().style.setProperty(
+            '--consonant-merch-card-mini-compare-top-section-mobile-height',
+            '32px',
+        );
+    }
+  }
+  adjustMiniCompareFooterRows () {
+    if (this.card.getBoundingClientRect().width === 0) return;
+    const footerRows = this.card.querySelector('[slot="footer-rows"]');
+    [...footerRows.children].forEach((el, index) => {
```

Comment:
nit: might be safer to use `...footerRows?.children` to avoid situations where the `footer-rows` element is not available.